server/src/domain/album/album.service.spec.ts

```diff
@@ -153,44 +155,71 @@ describe(AlbumService.name, () => {
   describe('create', () => {
     it('creates album', async () => {
       albumMock.create.mockResolvedValue(albumStub.empty);
-
-      await expect(sut.create(authStub.admin, { albumName: 'Empty album' })).resolves.toEqual({
+      userMock.get.mockResolvedValue(userStub.user1);
+      const expectedResult = {
         albumName: 'Empty album',
         description: '',
         albumThumbnailAssetId: null,
         assetCount: 0,
-        assets: [],
-        createdAt: expect.anything(),
+        assets: [
+          {
+            id: '123',
+            checksum: '',
+            deviceAssetId: '',
+            deviceId: '',
+            duration: '',
+            fileCreatedAt: new Date('1970-01-01'),
+            fileModifiedAt: new Date('1970-01-01'),
+            isArchived: false,
+            isFavorite: false,
+            originalFileName: '',
+            originalPath: '',
+            ownerId: '',
+            resized: false,
+            thumbhash: '',
+            type: AssetType.IMAGE,
+            updatedAt: new Date('1970-01-01'),
+          },
+        ],
+        createdAt: new Date('1970-01-01'),
         id: 'album-1',
-        owner: {
-          email: 'admin@test.com',
-          firstName: 'admin_first_name',
-          id: 'admin_id',
-          isAdmin: true,
-          lastName: 'admin_last_name',
-          oauthId: '',
-          profileImagePath: '',
-          shouldChangePassword: false,
-          storageLabel: 'admin',
-          createdAt: new Date('2021-01-01'),
-          deletedAt: null,
-          updatedAt: new Date('2021-01-01'),
-          externalPath: null,
-          memoriesEnabled: true,
-        },
+        owner: userStub.admin,
         ownerId: 'admin_id',
         shared: false,
-        sharedUsers: [],
+        sharedUsers: [userStub.user1],
         startDate: undefined,
         endDate: undefined,
         hasSharedLink: false,
-        updatedAt: expect.anything(),
-      });
+        updatedAt: new Date('1970-01-01'),
+      };
+      const mapAlbumWithAssetsSpy = jest.spyOn(albumResponse, 'mapAlbumWithAssets').mockReturnValue(expectedResult);
+
+      await expect(
+        sut.create(authStub.admin, {
+          albumName: 'Empty album',
+          sharedWithUserIds: ['user-id'],
+          description: '',
+          assetIds: ['123'],
+        }),
+      ).resolves.toEqual(expectedResult);
```

Comment:
That's correct at the moment but I figured it wouldn't be an disadvantage and in case someone were to modify the returned value at some point, it would catch that.
Basically I thought even a method like
```
foo(value) {
  return value
}
```
should be tested for the right return value in order to catch any unintended changes to the return value in the future.

Reply:
True, ish. I think the current test relies too much on internal implementation already though, was my main point.